src/js/hds-cc/hds-cc.js

```diff
@@ -1,41 +1,173 @@
+      bannerHeightElement: null,
+    };
+    this.cookie_name = 'city-of-helsinki-cookie-consents'; // Overridable default value
+
+    window.hds = window.hds || {};
+    window.hds.cookieConsent = this;
+
+    if (document.readyState === 'loading') {
+      document.addEventListener('DOMContentLoaded', () => {
+        this.#init();
+      });
+    } else {
+      this.#init();
+    }
+  }
+
+  /**
+   * Get the consent status for the specified cookie group names.
+   * @param {string[]} groupNamesArray - An array of group names.
+   * @return {Promise<boolean>} A promise that resolves to true if all the groups are accepted, otherwise false.
+   */
+  getConsentStatus(groupNamesArray) {
+
+    // Check if group names are provided as an array and not empty
+    if (!Array.isArray(groupNamesArray) || groupNamesArray.length === 0) {
+      console.error('Cookie consent: Group names must be provided as an non-empty array.');
+      return false;
+    }
+
+    const browserCookieState = this.#getCookie();
+
+    // Check if our cookie exists and has groups set
+    if (!browserCookieState || !browserCookieState.groups) {
+      // console.log('Cookie is not set properly', browserCookieState, browserCookieState.groups);
+      return false;
+    }
+
+    // Check if all groups are in accepted groups
+    return groupNamesArray.every(groupName => !!browserCookieState.groups[groupName]);
+  }
+
+  /**
+   * Sets the status of given cookie groups to accepted.
+   *
+   * @param {Array} acceptedGroupsArray - An array of cookie group names to be set as accepted.
+   * @return {Promise<boolean>} - A promise that resolves to true if the groups' status is successfully set to accepted, otherwise false.
+   */
+  async setGroupsStatusToAccepted(acceptedGroupsArray) {
+    const browserCookie = this.#getCookie();
+    const showBanner = browserCookie?.showBanner || false;
+
+    // If cookie is set, get the accepted groups
+    let currentlyAccepted = [];
+    if (browserCookie) {
+      currentlyAccepted = Object.keys(browserCookie.groups);
+    }
+
+    // Try to fetch the cookie settings from the JSON file
+    let cookieSettings = null;
+    try {
+      cookieSettings = await this.#getCookieSettingsFromJsonFile();
+    } catch (err) {
+      console.error(`Cookie consent: Unable to fetch cookie consent settings: ${err}`);
+      return false;
+    }
+
+    const cookieSettingsGroups = [...cookieSettings.requiredCookies.groups, ...cookieSettings.optionalCookies.groups];
+
+    // Checksums for all groups calculated in parallel without waiting for each
+    await Promise.all(cookieSettingsGroups.map(async (group) => {
+      group.checksum = await this.#getChecksum(group);
+    }));
 
-    // Debug helper. Open banner and run on console to see the box updated.
-    window.aaa_chatcheck = () => dispatchEvent(new CustomEvent('CONSENTS_CHANGED', { detail: { groups: ['chat'] } }));
+    // Save chat group to accepted groups and update checkbox state
```

Comment:
By use case this is chat group, but syntax wise should be 'acceptedGroupsArray'

Reply:
Fixed